packages/cli/src/models/files/ZosNetworkFile.ts

```diff
@@ -0,0 +1,365 @@
+import _ from 'lodash';
+import { Logger, FileSystem as fs, bytecodeDigest, bodyCode, constructorCode, semanticVersionToString } from 'zos-lib';
+import { fromContractFullName, toContractFullName } from '../../utils/naming';
+import { ZOS_VERSION, checkVersion } from './ZosVersion';
+import ZosPackageFile from './ZosPackageFile.js';
+import { ContractWrapper } from 'zos-lib';
+
+const log = new Logger('ZosNetworkFile');
+
+export default class ZosNetworkFile {
+
+  public packageFile: ZosPackageFile;
+  public network: any;
+  public fileName: string;
+  public data: any;
+
+  // TS-TODO: type for network parameter (and class member too).
+  constructor(packageFile: ZosPackageFile, network: any, fileName: string) {
+    this.packageFile = packageFile;
+    this.network = network;
+    this.fileName = fileName;
+
+    const defaults = { contracts: {}, solidityLibs: {}, proxies: {}, zosversion: ZOS_VERSION };
+
+    this.data = fs.parseJsonIfExists(this.fileName) || defaults;
+    checkVersion(this.data.zosversion, this.fileName);
+  }
+
+  get app(): any {
+    return this.data.app || {};
+  }
+
+  get appAddress(): string {
+    return this.app.address;
+  }
+
+  get package(): any {
+    return this.data.package || {};
+  }
+
+  get packageAddress(): string {
+    return this.package.address;
+  }
+
+  get provider(): any {
+    return this.data.provider || {};
+  }
+
+  get providerAddress(): string {
+    return this.provider.address;
+  }
+
+  get version(): string {
+    return this.data.version;
+  }
+
+  get frozen(): boolean {
+    return this.data.frozen;
+  }
+
+  get contracts(): any[] {
```

Comment:
nit: This will always return an object, not an array.
Also, this port if one of those where it would be extremely useful to define our own interfaces. For example, we could define a basic `ContractInterface` listing all its attributes without going very deep:
```ts
interface ContractsInterface {
  [contractAlias: string]: ContractInterface
}

interface ContractInterface {
  address: string;
  constructorCode: string;
  localBytecodeHash: string;
  ...
  types: any,
  storage: any[],
  ...
}

Reply:
Awesome!